src/main/java/com/checkmarx/intellij/tool/window/actions/selection/BranchSelectionGroup.java

```diff
@@ -42,14 +41,40 @@ public BranchSelectionGroup(@NotNull Project project,
     @Override
     protected @NotNull String getTitle() {
         if (getChildrenCount() == 0) {
-            return Bundle.message(Resource.BRANCH_SELECT_PREFIX) + ": " + (isEnabled() ? NONE_SELECTED : "...");
+            return Bundle.message(Resource.BRANCH_SELECT_PREFIX) + ": " + (isEnabled() ? setDefaultBranch() : "...");
         }
         String storedBranch = propertiesComponent.getValue(Constants.SELECTED_BRANCH_PROPERTY);
         return Bundle.message(Resource.BRANCH_SELECT_PREFIX)
                + ": "
-               + (StringUtils.isBlank(storedBranch) ? NONE_SELECTED : storedBranch);
+               + (StringUtils.isBlank(storedBranch) ? setDefaultBranch() : storedBranch);
     }
 
+    private String setDefaultBranch() {
+        if(branches == null || branches.isEmpty())  {
+            return NONE_SELECTED;
+        }
+        String activeBranch = getActiveBranch();
+        if(activeBranch == null) {
+            return NONE_SELECTED;
+        }
+        if(branches.contains(activeBranch)) {
+            updateActiveBranchAndFetchScans(activeBranch);
```

Comment:
Scenario:

Project1 is currently open in the CXOne IDE plugin.
Project2 is open in our main IDE.
Both projects have a branch named "mybranch".
Question: When the active branch in the IDE is set to "mybranch", will the IDE load the results from Project2 corresponding to its "mybranch" branch?

Expected Behavior: What should be the correct behavior in this situation? Specifically, should the IDE distinguish between branches with the same name across different projects?

Reply:
this is the reason that we want to give a warning to user when he tries to run a scan for CX project name other than the local project name